staramr/blast/results/plasmidfinder/PlasmidfinderHitHSP.py

```diff
@@ -0,0 +1,59 @@
+import logging
+import re
+
+from staramr.blast.results.AMRHitHSP import AMRHitHSP
+
+logger = logging.getLogger('PlasmidfinderHitHSP')
+
+"""
+A Class storing a PlasmidFinder-specific BLAST hit/HSP.
+"""
+
+class PlasmidfinderHitHSP(AMRHitHSP):
```

Comment:
I think we should add some unit tests for this class to make sure the regular expression is parsing out the sequence ids properly.  For example, taking a look at this file <https://bitbucket.org/genomicepidemiology/plasmidfinder_db/src/master/enterobacteriaceae.fsa>, it looks like some of the sequence ids are structured like:

```
>RepA_1_pKPC-CAV1321_CP011611
```

While others are structured like:

```
IncB/O/K/Z_1__CU928147
```

That is, one difference is that some have a single underscore `_` in front of the accession id `_CP011611`, while others have a double underscore `__CU928147`.

Defining a unit test class lets us group together all the assumptions we are making about how to break apart and interpret these sequence ids, which can be modified later to handle additional cases (in case we miss something).

You can create a new file named `test_PlasmidfinderHitHSP.py` in [staramr/tests/unit/blast/results/pointfinder](https://github.com/phac-nml/staramr/tree/befb44bda34e6691c0df744547146604971b3efb/staramr/tests/unit/blast/results/pointfinder).

For how to structure the unit test, you can take a look at [test_PointfinderHitHSP.py](https://github.com/phac-nml/staramr/blob/befb44bda34e6691c0df744547146604971b3efb/staramr/tests/unit/blast/results/pointfinder/test_PointfinderHitHSP.py) and [test_BlastHitPartitions.py](https://github.com/phac-nml/staramr/blob/befb44bda34e6691c0df744547146604971b3efb/staramr/tests/unit/blast/results/test_BlastHitPartitions.py).

A test case should be structured something like:

```python
def testParseSequenceId1(self):
    test_blast_record = build_blast_record('RepA_1_pKPC-CAV1321_CP011611')
    plasmid_hit_hsp = PlasmidfinderHitHSP('test_file', test_blast_record)
    
    self.assertEqual('RepA', plasmid_hit_hsp.get_amr_gene_name(), 'Did not parse correct gene name')
```

Now, `build_blast_record()` needs to be defined such that it builds an appropriate mock `blast_record` object passed to [PlasmidfinderHitHSP.py](https://github.com/phac-nml/staramr/blob/befb44bda34e6691c0df744547146604971b3efb/staramr/blast/results/plasmidfinder/PlasmidfinderHitHSP.py#L14) which contains just enough information to be usable for our tests.

For example, on [this line](https://github.com/phac-nml/staramr/blob/befb44bda34e6691c0df744547146604971b3efb/staramr/blast/results/plasmidfinder/PlasmidfinderHitHSP.py#L26) we call `get_amr_gene_id()`, which tries to access `blast_record['qseqid']` (see [this line](https://github.com/phac-nml/staramr/blob/befb44bda34e6691c0df744547146604971b3efb/staramr/blast/results/AMRHitHSP.py#L67)). So, one way to create the mock `blast_record` is to define it as a dictionary, setting:

```
blast_record = {}
blast_record['qseqid'] = 'RepA_1_pKPC-CAV1321_CP011611'
```

More complicated ways of creating mock objects (e.g., creating mock implementations of methods) can be done by using the [mock](https://docs.python.org/3/library/unittest.mock.html) libraries in Python.

I hope this gives you enough information to get started. Try picking some sequence ids from [this plasmidfinder file](https://bitbucket.org/genomicepidemiology/plasmidfinder_db/src/master/enterobacteriaceae.fsa) and creating some test cases for it (and then modifying the `re.search()` defined below to properly parse out the information from the sequence id).

Reply:
Added the unit tests in commit 9807f0d , I had to modify the filter method for the sequence ID so it would pass all the cases